### PR TITLE
Add faster Gamma

### PR DESCRIPTION
**Pull request by rohanbabbar04**
## Description
- Add class `Gamma` with all methods
- Add test in `test_scipy.py`.
- Update docs

## Checklist

- [x] Code style is correct (follows pylint and black guidelines)
- [x] Includes new or updated tests to cover the new feature
- [x] New features are properly documented (with an example if appropriate)

### CODE REVIEW

**Comment by rohanbabbar04** [comment]
One of my test in `test_maxent` is failing, I have commented it as of now

Any idea why is this happening?

File: preliz/tests/test_maxent.py [hunk 1]
```diff
@@ -78,7 +78,7 @@
         (ExGaussian(), 9, 10, 0.8, (-np.inf, np.inf), (9.496, 0.390, 0.003)),
         (ExGaussian(sigma=0.2), 9, 10, 0.8, (-np.inf, np.inf), (9.168, 0.423)),
         (Exponential(), 0, 4, 0.9, (0, np.inf), (0.575)),
-        (Gamma(), 0, 10, 0.7, (0, np.inf), (0.868, 0.103)),
+        # (Gamma(), 0, 10, 0.7, (0, np.inf), (0.868, 0.103)),
```
Comment (aloctavodia) [inline preliz/tests/test_maxent.py:81]: uncomment

File: preliz/distributions/gamma.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+@nb.njit(cache=True)
+def nb_entropy(alpha, beta):
+    return (
+        0.5 * (1.0 + np.log(2 * np.pi) + np.log(alpha))
+        - 1 / (3 * alpha)
+        - (alpha**-2.0) / 12
+        - (alpha**-3) / 90
+        + (alpha**-4) / 120
+    ) + np.log(1 / beta)
```
Comment (aloctavodia) [inline preliz/distributions/gamma.py:211]:
```suggestion
    return alpha - np.log(beta) + gammaln(alpha) + (1 - alpha) * digamma(alpha)
```

Import digamma from preliz.internal.special

File: preliz/distributions/gamma.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+    def rvs(self, size=None, random_state=None):
+        random_state = np.random.default_rng(random_state)
+        return random_state.gamma(self.alpha, 1 / self.beta, size)
+
+    def _fit_moments(self, mean, sigma):
+        alpha, beta = self._from_mu_sigma(mean, sigma)
+        self._update(alpha, beta)
+
+    def _fit_mle(self, sample, **kwargs):
```
Comment (aloctavodia) [inline preliz/distributions/gamma.py:178]:
```suggestion
    def _fit_mle(self, sample):
```

**Review by aloctavodia — CHANGES_REQUESTED** [review]

**Comment by aloctavodia** [comment]
The expression for the entropy was not correct, although it could return values close to the correct ones.

**Comment by rohanbabbar04** [comment]
> The expression for the entropy was not correct, although it could return values close to the correct ones.

Thanks, noted I forgot we have digamma already implemented in our `preliz.internal.special`

**Pull request merged**